pkgs/tools/security/tcplay/default.nix

```diff
@@ -0,0 +1,26 @@
+{ stdenv, fetchurl, cmake, pkgconfig, libgcrypt, libuuid, devicemapper, libudev, openssl, libgpgerror }:
+
+stdenv.mkDerivation rec {
+  name = "tcplay-2.0";
+
+  src = fetchurl {
```

Comment:
`fetchgit` does what you described.
`fetchFromGitHub` is just a wrapper around fetchzip and downloads the archive in tgz just like you manually do
It's declared [here](https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/all-packages.nix#L200)